### PR TITLE
C99 fixes

### PR DESCRIPTION
**Pull request by jacereda**
This PR adds support for compiling with `-std=c99`.

### CODE REVIEW

**Comment by jacereda** [comment]
If we decide in #855 that c89 is the way to go the fixes are still valid.

File: src/Sumtypes.hs [hunk 1]
```diff
@@ -345,6 +339,15 @@ copyCase typeEnv env concreteStructTy@(StructTy _ typeVariables) (theCase, isFir
   let (name, tys, correctedTagName) = namesFromCase theCase concreteStructTy
   in unlines
      [ "    " ++ (if isFirstCase then "" else "else ") ++ "if(pRef->_tag == " ++ correctedTagName ++ ") {"
-     , joinLines (map (memberCopy typeEnv env) (zip (map (\anon -> name ++ "." ++ anon) anonMemberNames) tys))
+     , joinLines $ memberCopy typeEnv env <$> unionMembers name tys
      , "    }"
      ]
+
+anonMemberName :: String -> String -> String
```
Comment (eriksvedang) [inline src/Sumtypes.hs:346]: What issue does this resolve? (mainly curious)

Comment (jacereda) [reply src/Sumtypes.hs:346]: Anonymous unions aren't allowed in C99. Sumtype cases are now accessed as `.u.memberX`.

Comment (eriksvedang) [reply src/Sumtypes.hs:346]: OK, I see. Thanks!

**Comment by eriksvedang** [comment]
Thanks!

**Pull request merged**